### PR TITLE
Remove Cloud API library

### PR DESCRIPTION
**Pull request by jayteemo**
Deprecate the Cloud API library.  Use the desired cloud service library directly instead of the Cloud API.
Remove the Cloud client sample, as it demonstrates the Cloud API.
Remove support for the Cloud API from nrf_cloud, aws_iot, and azure_iot_hub libraries.
Replace the Cloud API usage in serial_lte_modem application by directly using the nrf_cloud library.

### CODE REVIEW

**Comment by jayteemo** [comment]
it was discussed that the Cloud API should be removed.
remove now or mark as deprecated to be removed in a future release?
this PR does the former.
if the latter is desired, the last commit from the PR can be dropped.

**Comment by plskeggs** [comment]
Are their any tests either in sdk-nrf or in the Bitbucket test repos which need to change?  Forcing a test run should help check for this.

**Comment by jorgenmk** [comment]
@plskeggs Yes, there is a test for the cloud_client sample, but it's not part of the oncommit suite. Ie should not affect CI PR executions. I'll mark the test for removal.

**Comment by jorgenmk** [comment]
@plskeggs Yes, there is a test for the cloud_client sample, but it's not part of the oncommit suite. Ie should not affect CI PR executions. I'll mark the test for removal.

**Comment by jayteemo** [comment]
removed cloud api support from projects and marked cloud api as deprecated.

**Review by glarsennordic — APPROVED** [review]
Looks sane to me! I haven't reviewed the final commit, since we're thinking about skipping that one. Otherwise, these changes make sense to me.

**Review by plskeggs — APPROVED** [review]
Looks good to me.  Hopefully the sample was not highlighted in the seminar recently.

**Review by peknis — APPROVED** [review]

File: doc/nrf/libraries/networking/cloud.rst [hunk 1]
```diff
@@ -7,6 +7,10 @@ Cloud API
    :local:
    :depth: 2
 
+.. note::
+
+   The cloud API library is deprecated. Use the desired cloud service library directly.
```
Comment (coderbyheart) [inline doc/nrf/libraries/networking/cloud.rst:12]: Deprecation means that an API will be removed in the future, but customers can keep using it. However this change removes it without prior deprecation.

It seems to me that this change actually removes the cloud API?

**Review by jhn-nordic — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

File: doc/nrf/libraries/networking/cloud.rst (hunk 1 above)
Comment (jayteemo) [reply doc/nrf/libraries/networking/cloud.rst:12]: the cloud api library remains, but implementations of it have been removed.

**Comment by jayteemo** [comment]
updated to fully remove cloud API 😅

**Review by simensrostad — APPROVED** [review]
Goodbye, old friend.

**Pull request merged**